Indicators/KnowSureThing.cs

```diff
@@ -0,0 +1,219 @@
+        /// <param name="signalPeriod">The smoothing period used to smooth the signal values</param>
+        /// <param name="movingAverageType">Specifies the type of moving average to be used as smoother for KnowSureThing values</param>
+        public KnowSureThing(string name, int roc1Period, int roc1MaPeriod, int roc2Period, int roc2MaPeriod,
+            int roc3Period, int roc3MaPeriod, int roc4Period, int roc4MaPeriod, int signalPeriod,
+            MovingAverageType movingAverageType = MovingAverageType.Simple)
+            : base(name)
+        {
+            WarmUpPeriod = (new int[] {roc1Period + roc1MaPeriod, roc2Period + roc2MaPeriod, roc3Period + roc3MaPeriod, roc4Period + roc4MaPeriod }).Max();
+
+            MovingAverageType = movingAverageType;
+
+            ROC1 = new RateOfChange(roc1Period);
+            ROC2 = new RateOfChange(roc2Period);
+            ROC3 = new RateOfChange(roc3Period);
+            ROC4 = new RateOfChange(roc4Period);
+
+            ROC1MA = movingAverageType.AsIndicator(name + "_ROC1MA", roc1MaPeriod);
+            ROC2MA = movingAverageType.AsIndicator(name + "_ROC2MA", roc2MaPeriod);
+            ROC3MA = movingAverageType.AsIndicator(name + "_ROC3MA", roc3MaPeriod);
+            ROC4MA = movingAverageType.AsIndicator(name + "_ROC4MA", roc4MaPeriod);
+
+            SignalLine = movingAverageType.AsIndicator(name + "_SignalLine", signalPeriod);
+
+            KST = new Identity(name + "_KST");
+            Signal = new Identity(name + "_Signal");
+        }
+
+        /// <summary>
+        /// Computes the next value of this indicator from the given state
+        /// </summary>
+        /// <param name="input">The input given to the indicator</param>
+        /// <returns>The input is returned unmodified.</returns>
+        protected override decimal ComputeNextValue(IndicatorDataPoint input)
+        {
+            ROC1.Update(input);
+            ROC2.Update(input);
+            ROC3.Update(input);
+            ROC4.Update(input);
+
+            if (ROC1.IsReady)
+            {
+                ROC1MA.Update(input.EndTime, 100 * ROC1.Current.Value);
+            }
+            if (ROC2.IsReady)
+            {
+                ROC2MA.Update(input.EndTime, 100 * ROC2.Current.Value);
+            }
+            if (ROC3.IsReady)
+            {
+                ROC3MA.Update(input.EndTime, 100 * ROC3.Current.Value);
+            }
+            if (ROC4.IsReady)
+            {
+                ROC4MA.Update(input.EndTime, 100 * ROC4.Current.Value);
+            }
+
+            var kst = ROC1MA.Current.Value + 2 * ROC2MA.Current.Value + 3 * ROC3MA.Current.Value + 4 * ROC4MA.Current.Value;
+
+            SignalLine.Update(input.EndTime, kst);
+
+            if (!SignalLine.IsReady)
+            {
+                return 0m;
+            }
+
+            KST.Update(input.EndTime, kst);
+            Signal.Update(input.EndTime, SignalLine.Current.Value);
+
+            return input.Value;
```

Comment:
Think the indicator should return `kst` so it's used as the indicator value, no need for the KST `Identity` ?
Same for `Signal` it's value is the same as `SignalLine`, think it can just be removed